src/OrchardCore/OrchardCore.UrlRewriting.Core/Services/UrlRedirectRuleSource.cs

```diff
@@ -0,0 +1,95 @@
+using System.Text;
+using Microsoft.AspNetCore.Rewrite;
+using Microsoft.Extensions.Localization;
+using OrchardCore.Entities;
+using OrchardCore.Modules.Extensions;
+using OrchardCore.UrlRewriting.Models;
+
+namespace OrchardCore.UrlRewriting.Services;
+
+public sealed class UrlRedirectRuleSource : IUrlRewriteRuleSource
+{
+    public const string SourceName = "Redirect";
+
+    internal IStringLocalizer S;
+
+    public UrlRedirectRuleSource(IStringLocalizer<UrlRedirectRuleSource> stringLocalizer)
+    {
+        S = stringLocalizer;
+
+        Description = S["URL Redirect Rule"];
+    }
+
+    public string Name
+        => SourceName;
+
+    public LocalizedString Description { get; }
+
+    public void Configure(RewriteOptions options, RewriteRule rule)
+    {
+        if (!rule.TryGet<UrlRedirectSourceMetadata>(out var metadata) ||
+            string.IsNullOrEmpty(metadata.Pattern) ||
+            string.IsNullOrEmpty(metadata.SubstitutionUrl))
+        {
+            return;
+        }
+
+        var builder = new StringBuilder();
+
+        builder.Append("RewriteRule \"");
+        builder.Append(metadata.Pattern);
+        builder.Append("\" \"");
+        builder.Append(metadata.SubstitutionUrl);
+        builder.Append("\" [");
+
+        var builderFlags = new StringBuilder();
```

Comment:
@arkadiuszwojcik why did you user another string builder here? You should just use the same string builder instance as I did before the last change

Reply:
To simplify code, otherwise code looks the same as before inroducing `AppendCommaSeparatedValues` where initial comma have to be manualy tracked. Btw, isn't better to use `AppendFormat` for `Pattern` and `SubstitutionUrl`? This way string format would be much easier to read and overhead will be almost the same.

Reply:
@MikeAlhayek I changed implementation, please take a look is it ok for you

Reply:
yes this is okay still.